server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -338,6 +338,25 @@ public Consumer getConsumer(String uuid) {
         return this.cpQueryFactory.<Consumer>buildQuery(this.currentSession(), criteria);
     }
 
+    @SuppressWarnings("unchecked")
+    @Transactional
+    public CandlepinQuery<Consumer> listByRecipientOwner(Owner owner) {
+        DetachedCriteria criteria = this.createSecureDetachedCriteria()
+            .add(Restrictions.eq("recipientOwnerKey", owner.getKey()));
+
+        return this.cpQueryFactory.<Consumer>buildQuery(this.currentSession(), criteria);
+    }
+
+    public boolean doesShareConsumerExist(Owner owner) {
+        long result = (Long) createSecureCriteria()
+            .add(Restrictions.eq("owner", owner))
+            .add(Restrictions.isNotNull("recipientOwnerKey"))
+            .setProjection(Projections.count("id"))
+            .uniqueResult();
```

Comment:
`uniqueResult` returns null if there is not a result, not 0.  I would just drop `uniqueResult` entirely, grab the count, and `return count != 0`